### PR TITLE
Gracefully fail when freshness cannot be calculated

### PR DESCRIPTION
**Pull request by parkerhancock**
This is a simple change to remove a Runtime Error when a freshness can't be calculated. If freshness can't be calculated, then the request should be sent, rather than error out.

### CODE REVIEW

**Comment by karpetrosyan** [comment]
Thank you very much; that makes sense.

So RFC 9111 says:

> Caches are encouraged to consider responses that have invalid freshness information (e.g., a max-age directive with non-integer content) to be stale

That means we should re-validate the response instead of raising a `RuntimeError`.
I will add a changelog for this and merge it.

**Comment by karpetrosyan** [comment]
Can you please add a test and changelog for this PR?

Here is how to test it:

```python
def test_freshness_lifetime_invalid_information():
    controller = Controller()
    response = Response(
        status=200,
    )
    original_request = Request("GET", "https://example.com")
    request = Request("GET", "https://example.com")
    conditional_request = controller.construct_response_from_cache(
        request=request, response=response, original_request=original_request
    )
    assert isinstance(conditional_request, Request)
```

**Comment by parkerhancock** [comment]
Requested changes made! I also confirmed that the new test added actually tests for this change (I wrote the test, reverted the change to _controller.py, confirmed failing, restored change to _controller.py, test now passes).

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## 0.0.18 (9/11/2023)
+
+- Fix issue where freshness cannot be calculated to re-send request (#104)
```
Comment (karpetrosyan) [inline CHANGELOG.md:5]:
```suggestion
- Fix issue where freshness cannot be calculated to re-send request. (#104)
```

File: tests/test_controller.py [hunk 2]
```diff
@@ -773,3 +773,15 @@ def test_must_understand_response_directive():
     controller = Controller()
 
     assert controller.is_cachable(request=request, response=response)
+
```
Comment (karpetrosyan) [inline tests/test_controller.py:776]:
```suggestion


```

File: tests/test_controller.py [hunk 3]
```diff
@@ -773,3 +773,15 @@ def test_must_understand_response_directive():
...
+    controller = Controller()
+    response = Response(
+        status=400,
+    )
+    original_request = Request("GET", "https://example.com")
+    request = Request("GET", "https://example.com")
+    conditional_request = controller.construct_response_from_cache(
+        request=request, response=response, original_request=original_request
+    )
+    assert isinstance(conditional_request, Request)
```
Comment (karpetrosyan) [inline tests/test_controller.py:787]:
```suggestion
    assert isinstance(conditional_request, Request)

```

File: tests/test_controller.py [hunk 4]
```diff
@@ -773,3 +773,14 @@ def test_must_understand_response_directive():
     controller = Controller()
 
     assert controller.is_cachable(request=request, response=response)
+def test_freshness_lifetime_invalid_information():
```
Comment (karpetrosyan) [inline tests/test_controller.py:776]:
```suggestion


def test_freshness_lifetime_invalid_information():
```

File: tests/test_controller.py [hunk 5]
```diff
@@ -773,3 +773,16 @@ def test_must_understand_response_directive():
...
+    controller = Controller()
+    response = Response(
+        status=400,
+    )
+    original_request = Request("GET", "https://example.com")
+    request = Request("GET", "https://example.com")
+    conditional_request = controller.construct_response_from_cache(
+        request=request, response=response, original_request=original_request
+    )
+    assert isinstance(conditional_request, Request)
```
Comment (karpetrosyan) [inline tests/test_controller.py:788]:
```suggestion
    assert isinstance(conditional_request, Request)

```

**Comment by karpetrosyan** [comment]
Great job! Thank you @parkerhancock

**Pull request merged**